### PR TITLE
Beacon api: produce block should skip mev-boost

### LINKED ISSUE #11481: Fetching beacon block through API triggers relay request
# 🐞 Bug Report

### Description

When an attempt to obtain a beacon block via the `/eth/v2/validator/blocks/{slot}` is made to a beacon node configured with an MEV relay, the beacon node erroneously contacts that MEV relay.

### Has this worked before in a previous version?

No.

## 🔬 Minimal Reproduction

- configure prysm with the `--http-mev-relay` option
- attempt to fetch a block via the REST API with the `/eth/v2/validator/blocks/{slot}` endpoint

## 🔥 Error

A call to ` /eth/v2/validator/blocks/{slot}` should not attempt to contact MEV relays.  This call must return a complete beacon block proposal, which requires a complete execution payload.  It cannot obtain this from the MEV relay at this point in the process, so instead should obtain a complete execution payload from its connected execution node.

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**

When calling beacon API `/eth/v2/validator/blocks/{slot}` and `/eth/v2/validator/blocks/{slot}/ssz`, these end points should not call mev-boost/relayer to outsource block construction. This PR fixes it by adding field `SkipMevBoost` to  `BlockRequest` such that if `SkipMevBoost` is true, the prysm API server will not attempt to reach mev-boost. This is the cleanest solution IMO, the other options are to refactor prysm API's get beacon block into two methods where one calls mev-boost and one doesn't

**Which issues(s) does this PR fix?**

Fixes #11481

**Other notes for review**

Given that this is a protobuf change, we'll need to be careful there's no backward compatibility issue between beacon node and validator.

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go [hunk 1]
```diff
@@ -44,22 +44,24 @@ func (vs *Server) getBellatrixBeaconBlock(ctx context.Context, req *ethpb.BlockR
...
+			log.WithFields(logrus.Fields{
+				"slot":           req.Slot,
+				"validatorIndex": altairBlk.ProposerIndex,
+			}).Errorf("Could not determine validator has registered. Default to local execution client: %v", err)
 		}
-	} else if err != nil {
-		log.WithFields(logrus.Fields{
-			"slot":           req.Slot,
-			"validatorIndex": altairBlk.ProposerIndex,
-		}).Errorf("Could not determine validator has registered. Default to local execution client: %v", err)
```
Comment (prestonvanloon) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go:62]: Please add tests for these changes

File: beacon-chain/rpc/eth/validator/validator.go [hunk 2]
```diff
@@ -334,6 +337,8 @@ func (vs *Server) ProduceBlockV2(ctx context.Context, req *ethpbv1.ProduceBlockR
 // ProduceBlockV2SSZ requests the beacon node to produce a valid unsigned beacon block, which can then be signed by a proposer and submitted.
 //
 // The produced block is in SSZ form.
+// By definition `/eth/v2/validator/blocks/{slot}/ssz`, does not produce block using mev-boost and relayer network:
+// https://ethereum.github.io/beacon-APIs/?urls.primaryName=v2.3.0#/Validator/produceBlockV2
```
Comment (prestonvanloon) [inline beacon-chain/rpc/eth/validator/validator.go:341]: Please add a suggestion to which API endpoint is supposed to be used for MEV boost / relay. I.e. blinded blocks endpoint. 

The link you provided doesn't define anything about whether or not mev-boost should be used.

**Review by prestonvanloon — CHANGES_REQUESTED** [review]

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go:62]: This has been tested. The unit test is `TestServer_GetBellatrixBeaconBlock_BuilderCase`
Keep in mind that `req.SkipMevBoost ` is false by default

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go:62]: `req.SkipMevBoost` is true is tested through `TestProduceBlockV2SSZ`  and `TestProduceBlockV2`.
`req.SkipMevBoost` is false is tested through `TestServer_GetBellatrixBeaconBlock_BuilderCase`. Local test case.

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go:62]: Also added another local test `TestServer_GetBellatrixBeaconBlock_LocalProgressingWithBuilderSkipped`

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go [hunk 3]
```diff
@@ -44,22 +44,24 @@ func (vs *Server) getBellatrixBeaconBlock(ctx context.Context, req *ethpb.BlockR
...
+					"back to local execution client")
+				builderGetPayloadMissCount.Inc()
+			} else if builderReady {
+				return b, nil
+			}
+		} else if err != nil {
+			log.WithFields(logrus.Fields{
+				"slot":           req.Slot,
+				"validatorIndex": altairBlk.ProposerIndex,
+			}).Errorf("Could not determine validator has registered. Default to local execution client: %v", err)
```
Comment (prestonvanloon) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go:63]:
```suggestion
			log.WithError(err).WithFields(logrus.Fields{
				"slot":           req.Slot,
				"validatorIndex": altairBlk.ProposerIndex,
			}).Error("Could not determine validator has registered. Default to local execution client)
```

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**